src/widgets.ts

```diff
@@ -125,6 +129,29 @@ export abstract class Widget<
     })
   }
 
+  handleOAuthRedirect(redirectURL: string): void {
+    try {
```

Comment:
What in code in this method throws?

Reply:
`new URL` with throw if it is given a "bad" url.